### ISSUE TITLE
[BUG] Unexpected normalisation of DOS results

### ISSUE THREAD

**Issue by MBartkowiakSTFC** [ISISNeutronMuon/MDANSE#438]
**Description of the error**
I have run the Density of States on a trajectory of SrTiO3. Here are the results, depending on the atom selection:
![SrTiO3_DOS_allSrAtoms](https://github.com/ISISNeutronMuon/MDANSE/assets/108934199/ac000eef-ec95-47b5-b784-ed1fb95a0e4c)
Each one of those curves has been calculated for ALL the Sr atoms. The selection was:
1. blue: all atoms (Sr, Ti, O)
2. orange: all Sr and Ti atoms
3. green: all Sr and 4 Ti atoms,
4. red: all Sr and 1 Ti atom.

Similarly, the Ti results are not what I expected:
![SrTiO3_DOS_differentTiAtoms](https://github.com/ISISNeutronMuon/MDANSE/assets/108934199/1fc59b4b-959a-4351-8f53-e8cd491d0ce7)
By deselecting the O atoms I made the Ti signal increase.

**Describe the expected result**
I would prefer if the result for Sr atoms did not change when the selection of other atoms is changed.

**Describe the actual result**
At the moment, changing anything in the atom selection affects all the results.

**Suggested fix**
In the first approximation, we could normalise the results to ALL the atoms in the system, and not just the selected ones.

**Additional details**
N/A

**Comment by ChiCheng45** [comment]
I guess the weight scheme should be like this then

![image](https://github.com/ISISNeutronMuon/MDANSE/assets/14276033/67480d1a-255c-401c-b6b8-9a1d63c546c4)

where $a$ is the atom, $i$ is the index of an atom $a$ and $n_i$ is whether the atom with index $i$ is selected or not.

Currently I think its something like this

![image](https://github.com/ISISNeutronMuon/MDANSE/assets/14276033/48121c16-2d2a-483a-9da9-b8181087db43)

so the selection changes the normalization.

**Comment by MBartkowiakSTFC** [comment]
Yes, this would be my preference. Probably not a big deal since we are working with arbitrary units here, but still it would make the results more consistent.
